pkg/config/config.go

```diff
@@ -929,7 +929,31 @@ func loadDefaultEnv(env string) error {
 
 func loadEnvIfExists(path string) error {
 	if err := godotenv.Load(path); err != nil && !errors.Is(err, os.ErrNotExist) {
-		return errors.Errorf("failed to load %s: %w", ".env", err)
+		// If DEBUG=1, return the error as is for full debugability
+		if os.Getenv("DEBUG") == "1" {
+			return errors.Errorf("failed to load .env: %w", err)
+		}
+		msg := err.Error()
+		switch {
+		case strings.Contains(msg, "unexpected character"):
+			// Try to extract the character, fallback to generic
+			start := strings.Index(msg, "unexpected character \"")
+			if start != -1 {
+				start += len("unexpected character \"")
+				end := strings.Index(msg[start:], "\"")
+				if end != -1 {
+					char := msg[start : start+end]
+					return errors.Errorf("failed to parse environment file: .env (unexpected character '%s' in variable name)", char)
+				}
+			}
+			return errors.Errorf("failed to parse environment file: .env (unexpected character in variable name)")
+		case strings.Contains(msg, "unterminated quoted value"):
+			return errors.Errorf("failed to parse environment file: .env (unterminated quoted value)")
+		case strings.Contains(msg, "\n"):
```

Comment:
any explanation why we are checking for new lines?

Reply:
The sanitizing should handle the current possible parsing error (from the code I read from the godotenv lib). But if some update introduce new errors, this might break.

When the error message contain newlines, chances are, the newlines are actually the dotenv values being leaked away. Will add a comment for context though.